master/pkg/searcher/actions.go

```diff
@@ -0,0 +1,73 @@
+package searcher
+
+import (
+	"fmt"
+
+	"github.com/determined-ai/determined/master/pkg/model"
+
+	"github.com/determined-ai/determined/master/pkg/nprand"
+)
+
+// Action is an action that a searcher would like to perform.
+type Action interface {
+	SearcherAction()
```

Comment:
i think this can be unexported
```suggestion
	searcherAction()
```